ctest-next/src/lib.rs

```diff
@@ -31,3 +31,77 @@ pub type Error = Box<dyn std::error::Error>;
 pub type Result<T, E = Error> = std::result::Result<T, E>;
 /// A boxed string for representing identifiers.
 type BoxStr = Box<str>;
+
+/// A kind of item to which the C volatile qualifier could apply.
+///
+/// This is necessary because `ctest` does not parse the header file, so it
+/// does not know which items are volatile.
+#[derive(Debug)]
+#[non_exhaustive]
+pub enum VolatileItemKind {
+    /// A struct field.
+    StructField(Struct, Field),
+    /// An extern static.
+    Static(Static),
+    /// A function argument.
+    FnArgument(Fn, Box<Parameter>),
+    /// Function return type.
+    FnReturnType(Fn),
+}
+
+/// The kind of type a C type can be.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+#[non_exhaustive]
+pub enum TyKind {
+    /// A struct type.
+    Struct,
+    /// A union type.
+    Union,
+    /// Any other type.
+    Other,
+}
+
+/// Inputs needed to rename or skip a field.
+#[derive(Debug, Clone)]
+pub(crate) enum MapInput<'a> {
+    Struct(&'a Struct),
+    Fn(&'a crate::Fn),
+    #[expect(unused)]
+    Field(&'a Struct, &'a Field),
+    Alias(&'a Type),
+    Const(&'a Const),
+    Static(&'a Static),
+    Type(&'a str, TyKind),
+}
+
+// The From impls make it easier to write code in the test templates.
```

Comment:
```suggestion
/* The From impls make it easier to write code in the test templates. */
```

Nit, there's a casual convention to use `/* ... */` for section heading comments